src/utils/transactions.ts

```diff
@@ -84,3 +84,86 @@ export const useTransactionInfo = (tx: Transaction | MempoolTransaction, address
     lockTime
   }
 }
+
+// TODO: The following 2 functions could be ported to js-sdk (and properly tested there)
+type AttoAlphAmount = string
+type Address = string
+
+type UTXO = {
+  attoAlphAmount?: AttoAlphAmount
+  address?: Address
+  tokens?: { id: string; amount: string }[]
+}
+
+export const sumUpAlphAmounts = (utxos: UTXO[]): Record<Address, AttoAlphAmount> => {
+  const validUtxos = utxos.filter((utxo) => utxo.address && utxo.attoAlphAmount)
+
+  const grouped = groupBy(validUtxos, 'address')
+  const summed = mapValues(grouped, (addressGroup) =>
+    reduce(addressGroup, (sum, utxo) => (BigInt(sum) + BigInt(utxo.attoAlphAmount || 0)).toString(), '0')
+  )
+
+  return summed
+}
+
+export const sumUpTokenAmounts = (utxos: UTXO[]): Record<Address, Record<string, string>> => {
+  const validUtxos = utxos.filter((utxo) => utxo.address && utxo.tokens && utxo.tokens.length > 0)
+
+  const grouped = groupBy(validUtxos, 'address')
+  const summed = mapValues(grouped, (addressGroup) => {
+    const tokenSums: Record<string, string> = {}
+    for (const utxo of addressGroup) {
+      for (const token of utxo.tokens || []) {
+        if (!tokenSums[token.id]) {
+          tokenSums[token.id]
+        }
+        tokenSums[token.id] = (BigInt(tokenSums[token.id] || 0) + BigInt(token.amount)).toString()
+      }
+    }
+    return tokenSums
+  })
+
+  return summed
+}
+
+export const IOAmountsDelta = (
+  inputs: UTXO[] = [],
+  outputs: UTXO[] = []
+): { alph: Record<string, string>; tokens: Record<string, Record<string, string>> } => {
```

Comment:
nit: Since we have the `Address`, `AttoAlphAmount` and `Asset['id']` (from `web3`), wouldn't it be nicer to have better typing here than `string`?

Reply:
It would, definitely. I have to say that I was first focused to make it work quickly, but I'm now gladly adding more precise typings.